client/src/boot/applyConditionals.js

```diff
@@ -0,0 +1,26 @@
+import Injector from 'lib/Injector';
+
+const applyConditionals = () => {
+  Injector.transform(
+    'add-to-campaign',
+    (updater) => {
+      updater.form.alterSchema(
+        '*.AddToCampaign',
+        (form) => {
+          const toggle = form.getFieldByName('AddNewSelect');
+          if (!toggle) {
+            return form.getState();
+          }
+          const addNewField = form.getFieldByName('NewTitle');
+          const visible = form.getValue('AddNewSelect') || addNewField.message;
```

Comment:
this condition leaves the UI is an inconsistent state

Could the server force the checkbox value to visible if there is a message instead?

Having a message on `addNewField` makes that checkbox completely redundant...
Alternatively, hide the checkbox if there is a message, whichever is easier.